lib/virtual.py

```diff
@@ -0,0 +1,144 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2024 IBM
+# Author: Shaik Abdulla <abdulla1@linux.vnet.ibm.com>
+
+"""
+Module for all Virtual devices related functions.
+"""
+
+import re
+import os
+import platform
+
+from .helper import runcmd
+
+
+def get_mac_address(interface):
+    '''
+    Gets the mac_address of given interface.
+    parameter: interface: Name of Interface.
+
+    :return: string of MAC address.
+    '''
+    try:
+        for line in runcmd("ip a s dev %s" % interface)[1].splitlines():
+            if 'link/ether' in line:
+                mac_address = line.split()[1]
+                return mac_address
+    except Exception as e:
+        print(f'Interface not found {e}')
+
+
+def get_driver(interface):
+    '''
+    Gets associated driver/module of given interface.
+    parameter: interface: Name of Interface.
+
+    :return: string of driver name.
+    '''
+    for line in runcmd("ethtool -i  %s" % interface)[1].splitlines():
+        if line.startswith('driver:'):
+            driver = line.split(': ')[1].strip()
+            return driver
+
+def get_veth_interface_names():
+    '''
+    Gets all vETH interface names.
+
+    :return: list of virtual interface names.
+    '''
+    veth_list = []
+    for input_string in runcmd("lsdevinfo -c")[1].splitlines():
```

Comment:
is `lsdevinfo` , `ip` and other commands available default in the system, better to check and install else proper error handling

Reply:
@PraveenPenguin The interface extraction specific to vNIC and vETH, "lsdevinfo" utility is very handy to extract it.
implemented error catch handling for same.